### PR TITLE
[tests-only] [full-ci] removing the setresponse in given/then step in spaces context

### PR DESCRIPTION
**Pull request by KarunAtreya**
## Description
We have used setResponse() and $this->response in the Given/Then steps and some helper functions (maybe to reuse existing available methods). But storing responses from Given/Then steps and helper functions is not a good idea because it can lead to a false positive assertion in the Then steps.
So, check the use of setResponse() and $this->response in
- Given steps
- Then steps (Then steps can use $this->response but must prevent saving to it)
- Helper functions

So this pr make the above changes in `SpacesContext`
## Related Issue
https://github.com/owncloud/ocis/issues/7082

## Motivation and Context
- To  remove setResponse() and $this->response in the Given/Then steps and some helper functions
- To avoid false positive assertions

## How Has This Been Tested?

- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 1]
```diff
@@ -514,21 +537,33 @@ public function sendCreateFolderRequest(
...
 	 *
+	 * @param string $user
+	 * @param string $query
+	 *
+	 * @return ResponseInterface
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function listAllAvailableSpacesUsingTheGraphApi(string $user, string $query = ''): ResponseInterface {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:554]:
```suggestion
	public function listAllAvailableSpaces(string $user, string $query = ''): ResponseInterface {
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -501,6 +503,27 @@ public function sendCreateFolderRequest(
...
+	/**
+	 * @param string $user
+	 * @param string $query
+	 *
+	 * @return ResponseInterface
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function listAllAvailableSpacesOfUserUsingTheGraphApi(string $user, string $query = ''): ResponseInterface {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:515]:
```suggestion
	public function listAllAvailableSpacesOfUser(string $user, string $query = ''): ResponseInterface {
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 3]
```diff
@@ -159,16 +159,18 @@ public function getSpaceIdByNameFromResponse(string $name): string {
...
 	 * @throws Exception
 	 */
-	public function getSpaceByNameFromResponse(string $name): array {
-		$response = json_decode((string)$this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
-		$spaceAsArray = $response;
-		if (isset($response['name']) && $response['name'] === $name) {
-			return $response;
+	public function getSpaceByNameFromResponse(string $name, ?ResponseInterface $response = null): array {
+		$response = $response ?? $this->featureContext->getResponse();
+		$decodedResponse = json_decode((string)$response->getBody(), true, 512, JSON_THROW_ON_ERROR);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:170]: will function `getJsonDecodedResponseBodyContent()` work here?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 4]
```diff
@@ -970,14 +999,15 @@ public function theUserShouldHaveASpaceInTheDisableState(
...
 		$this->featureContext->theHTTPStatusCodeShouldBe(
 			200,
-			"Expected response status code should be 200"
+			"Expected response status code should be 200",
+			$response
 		);
-		$response = json_decode((string)$this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
-		if (isset($response["value"])) {
-			foreach ($response["value"] as $spaceCandidate) {
+		$decodedResponse = json_decode((string)$response->getBody(), true, 512, JSON_THROW_ON_ERROR);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1008]: will function `getJsonDecodedResponseBodyContent()` work here?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 5]
```diff
@@ -1439,15 +1471,40 @@ public function updateSpaceSpecialSection(
...
+	 * @param string $user
+	 * @param string $file
+	 * @param string $type
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userSetsFileAsImageInSpecialSectionOfSpace(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1496]:
```suggestion
	public function userSetsFileAsDescriptionOrImageInSpecialSectionOfSpace(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 6]
```diff
@@ -1578,22 +1638,20 @@ public function userCopiesFileWithinSpaceUsingTheWebDAVAPI(
...
 	 * @param string $fileSource
 	 * @param string $fileDestination
 	 * @param string $spaceName
 	 *
-	 * @return void
+	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public function userMovesFileWithinSpaceUsingTheWebDAVAPI(
+	public function moveFileWithinSpaceUsingTheWebDAVAPI(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1649]:
```suggestion
	public function moveFileWithinSpace(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 7]
```diff
@@ -1982,17 +2060,36 @@ public function createShareResource(
...
+	 * @When /^user "([^"]*)" creates a share inside of space "([^"]*)" with settings:$/
+	 *
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userCreatesShareInsideSpace(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2085]:
```suggestion
	public function userCreatesShareInsideSpaceWithSettings(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 8]
```diff
@@ -2106,18 +2198,37 @@ public function createPublicLinkToEntityInsideOfSpaceRequest(
...
+	 * @When /^user "([^"]*)" creates a public link share inside of space "([^"]*)" with settings:$/
+	 *
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userCreatesPublicLinkShareInsideOfSpace(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2224]:
```suggestion
	public function userCreatesPublicLinkShareInsideOfSpaceWithSettings(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 9]
```diff
@@ -2184,38 +2297,78 @@ public function userHasUnsharedASpaceSharedWith(
...
+
+	/**
+	 * @param  string $user
+	 * @param  string $object
+	 * @param  string $spaceName
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public function sendRemoveObjectFromSpaceRequest(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2360]: and what is this object? file/folder?
```suggestion
	public function removeObjectFromSpace(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 10]
```diff
@@ -2259,10 +2405,34 @@ public function userHasDeletedASpaceOwnedByUser(
...
+
+	/**
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param  string $owner
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public function sendDisableSpaceRequest(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2424]:
```suggestion
	public function disableSpace(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 11]
```diff
@@ -2330,11 +2494,36 @@ public function sendUserHasDisabledSpaceRequest(
...
+
+	/**
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param string $owner
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public function sendDeleteSpaceRequest(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2514]:
```suggestion
	public function deleteSpace(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 12]
```diff
@@ -2349,21 +2538,36 @@ public function sendUserHasDisabledSpaceRequest(
...
+
+	/**
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param  string $owner
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public function sendRestoreSpaceRequest(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2559]:
```suggestion
	public function restoreSpace(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 13]
```diff
@@ -2408,11 +2605,34 @@ public function userHasRestoredSpaceRequest(
...
+	}
+
+	/**
+	 * @param  string $user
+	 * @param  string $spaceName
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public function listAllDeletedFilesInTrash(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2624]:
```suggestion
	public function listAllDeletedFilesFromTrash(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 14]
```diff
@@ -2944,18 +3161,41 @@ public function sendShareSpaceViaLinkRequest(
...
+	 * @When /^user "([^"]*)" creates a public link share of the space "([^"]*)" with settings:$/
+	 *
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param TableNode|null $table
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userCreatesAPublicLinkShareOfSpace(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3187]:
```suggestion
	public function userCreatesAPublicLinkShareOfSpaceWithSettings(
```

File: tests/acceptance/features/bootstrap/SpacesContext.php (hunk 9 above)
Comment (KarunAtreya) [reply tests/acceptance/features/bootstrap/SpacesContext.php:2360]: both

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/SpacesContext.php:2360]: then `removeResourceFromSpace` would fit as the function name

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Pull request merged**